src/gluonts/mx/distribution/genpareto.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+import math
+from functools import partial
+from typing import Dict, List, Optional, Tuple
+
+# Third-party imports
+import numpy as np
+
+from gluonts.core.component import validated
+
+from gluonts.model.common import Tensor
+from .distribution import Distribution
+
+from gluonts.mx.distribution import Distribution
+from gluonts.mx.distribution.distribution import (
+    getF,
+    softplus,
+    _sample_multiple,
+)
+from gluonts.mx.distribution import uniform, box_cox_transform
+from gluonts.mx.distribution.distribution_output import DistributionOutput
+
+
+class GenPareto(Distribution):
+    r"""
+    Generalised Pareto distribution.
+
+    Parameters
+    ----------
+    xi
+        Tensor containing the xi shape parameters, of shape `(*batch_shape, *event_shape)`.
+    beta
+        Tensor containing the beta scale parameters, of shape `(*batch_shape, *event_shape)`.
+    loc
+        Tensor containing the (for-now positive) location parameters, of shape `(*batch_shape, *event_shape)`.
```

Comment:
Why for now positive? It seems like the corresponding `GenParetoOutput` class doesn’t constrain `loc`

Reply:
To your point about including a test for MLE via SGD, I dropped the inclusion of a location-shift parameter in order to permit this test's inclusion. MLE with a location-shift parameter requires Grimshaw's algorithm [https://amstat.tandfonline.com/doi/abs/10.1080/00401706.1993.10485040], and can perhaps be a future commit.